experimental/shared/src/main/scala/zio/prelude/experimental/EitherCompose.scala

```diff
@@ -0,0 +1,62 @@
+package zio.prelude
+package experimental
+
+trait EitherCompose[:=>[-_, +_]] extends AssociativeCompose[:=>] {
```

Comment:
Why not `ComposeEither`, `ComposeBoth`?

Reply:
To keep things regular. It's always `*Compose`, like `AssociativeCompose` or `IdentityCompose`. It is like this for all other type class names, like `*Both`, `*Either`, `*Flatten`, etc.
Flipping the order just here in this case would most likely only confuse people.

Reply:
I actually think this cuts the other way. When looking for relevant type classes for something it is nice to have things grouped together. If we had `ComposeEither`, `ComposeBoth`, and so on it would be easy to know where you should look for type classes for "function like" things and that those are different from type classes for other things.

Reply:
:thinking: are you suggesting that we rename `AssociativeCompose` -> `ComposeAssociative` and `IdentityCompose` -> `ComposeIdentity`?

Reply:
That's at least an idea.

Reply:
I think it would be better to change the naming scheme in a dedicated PR
https://github.com/zio/zio-prelude/issues/479